src/ProxyManager/ProxyGenerator/AccessInterceptorScopeLocalizer/MethodGenerator/BindProxyProperty.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ProxyManager\ProxyGenerator\AccessInterceptorScopeLocalizer\MethodGenerator;
+
+use Laminas\Code\Generator\ParameterGenerator;
+use ProxyManager\Exception\UnsupportedProxiedClassException;
+use ProxyManager\Generator\MethodGenerator;
+use ReflectionClass;
+
+use function strtr;
+
+class BindProxyProperty extends MethodGenerator
+{
+    public function __construct(ReflectionClass $originalClass)
+    {
+        parent::__construct(
+            'bindProxyProperty',
+            [
+                new ParameterGenerator('localizedObject', $originalClass->getName()),
+                new ParameterGenerator('class', ReflectionClass::class),
+                new ParameterGenerator('propertyName', 'string'),
+            ],
+            self::FLAG_PRIVATE
+        );
+
+        $bodyTemplate = <<<'CODE'
+        $originalClass = $class;
+        while (! $class->hasProperty($propertyName)) {
+            $class = $class->getParentClass();
+        }
+        $property = $class->getProperty($propertyName);
+        $property->setAccessible(true);
+        if (!$property->isInitialized($localizedObject)) {
```

Comment:
If I understand this correctly, the difference here is that we aren't creating an "empty shell", but proxying when the other class is correctly initialized?

Reply:
Yes: for now proxy can be created only when properties are nullable and have default value in class itself.
Thus this library tries to be responsible for user classes and objects.
This PR moves responsibility of initializing object to the user of the library: class may have non-nullable typed properties, but as long as concrete instance being proxied is properly initialized - it can be proxied.